DEVELOPMENT.md

```diff
@@ -54,3 +54,155 @@ For an example of this, refer to the [`rust.yaml` Github Workflow](./.github/wor
 [binstall]: https://github.com/cargo-bins/cargo-binstall
 [node.js]: https://nodejs.org/en/learn/getting-started/how-to-install-nodejs
 [protobuf-compiler]: https://grpc.io/docs/protoc-installation/
+
+## Makefile Overview
+
+This `Makefile` coordinates the build process for a Rust and TypeScript project using Docker. It integrates various development tools and dependencies for efficient multi-environment builds. Below are key targets and variables.
+
+### Variables
+
+- **`BASE_IMAGE_NAME`**: Base Docker image name.
+- **`COMMON_MONOLITH_IMAGE_NAME`**: Name for the monolithic Docker image.
+- **`COMMON_BUILDER_IMAGE_NAME`**: Builder Docker image name.
+- **`COMMON_RUNTIME_IMAGE_NAME`**: Runtime Docker image name.
+- **`RUNTIME_BASE_IMAGE`**: Base image for the runtime environment (default: Debian latest).
+- **`TOOLCHAIN`**: Rust toolchain version.
+- **`NODE_VERSION`**: Node.js version (default: latest LTS).
+- **`PLATFORM`**: Build platform (default: Ubuntu latest).
+- **`RUST_BACKTRACE`**: Enables Rust backtraces.
+- **`WIREIT_LOGGER`**: Configures WireIt logging (default: metrics).
+- **`DOCKER_BUILDKIT`**: Enables Docker BuildKit.
+- **`COMPOSE_DOCKER_CLI_BUILD`**: Enables BuildKit for Docker Compose.
+
+### Targets
+
+#### `all`
+
+- **Description**: Default target that triggers the local build process.
+- **Dependencies**: `local`
+
+#### `local`
+
+- **Description**: Builds the Rust project and lists the build outputs.
+- **Dependencies**: `build_rust`, `list_outputs`
+
+#### `docker`
+
+- **Description**: Builds all Docker images.
+- **Dependencies**: `docker_base`, `docker_monolith`, `docker_builder`, `docker_runtime`
+
+#### `docker_base`
+
+- **Description**: Builds the base Docker image.
+- **Dependencies**: `.docker_base.done`
+
+#### `docker_monolith`
+
+- **Description**: Builds the monolithic Docker image.
+- **Dependencies**: `.docker_monolith.done`, `docker_base`
+
+#### `docker_builder`
+
+- **Description**: Builds the Docker image for the build process.
+- **Dependencies**: `.docker_builder.done`, `docker_base`
+
+#### `docker_runtime`
+
+- **Description**: Builds the Docker image for the runtime environment.
+- **Dependencies**: `.docker_runtime.done`, `docker_base`
+
+#### `build_typescript`
+
+- **Description**: Installs npm dependencies and builds the TypeScript project.
+- **Dependencies**: `.build_typescript.done`
+
+#### `build_rust`
+
+- **Description**: Builds the Rust project in release mode.
+- **Dependencies**: `.build_rust.done`
+
+#### `list_outputs`
+
+- **Description**: Lists the outputs of the Rust build.
+- **Dependencies**: `.build_rust.done`
+
+### Phony Targets
+
+These targets do not correspond to files and are always executed when invoked.
+The `.done` files are used to track the completion of build steps.
+
+- `all`
+- `local`
+- `docker`
+- `docker_base`
+- `docker_monolith`
+- `docker_builder`
+- `docker_runtime`
+- `build_typescript`
+- `build_rust`
+- `list_outputs`
+
+### Usage
```

Comment:
Can we move `###Usage` up directly below `##Makefile Overview`?

It feels weird to me that usage isn't right underneath the overview. As a reader, I want to read the overview and then immediately see a few example commands I can run, rather than go hunting for them. The rest of this feels like it makes good reference material to follow on from that.

Reply:
- see https://github.com/commontoolsinc/system/pull/73/commits/92d61a986bdb0a9c827c286a09a4e44ca1780879